File: C++/869. Reordered Power of 2/lc869.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n) {
        vector <long> power2list;
        for( int i=0;i<32;i++)
        {
            power2list.push_back(pow(2,i));
        }
        vector <string> power2StrList;
        for (auto x : power2list)
        {
            string xStr=to_string(x);
            sort(xStr.begin(),xStr.end());                
            power2StrList.push_back(xStr);
        }
        string nStr= to_string(n);
        for (auto s: power2StrList )
        {
            sort(nStr.begin(),nStr.end());
            if(nStr==s)
                return true;
        }
        return false;
    }
};

File: Python/869. Reordered Power of 2/lc869.py
class Solution:
    def reorderedPowerOf2(self, n: int) -> bool:
        power2List=[pow(2,x) for x in range(31) ]
        pow2StrList=[sorted(str(x)) for x in power2List]
        nStr=sorted(str(n))
        if nStr in pow2StrList:
            return True
        return False
